### ISSUE TITLE
graphql-server-typescript example does not work

### ISSUE THREAD

**Issue by prishub** [accounts-js/accounts#806]
# graphql-server-typescript example does not work

## Description

Page https://github.com/accounts-js/accounts/tree/master/examples/graphql-server-typescript states "This example demonstrate how to use accounts-js."

Following instructions...

```bash
git clone https://github.com/accounts-js/examples.git
cd examples/graphql-server-typescript
yarn install
```

...leads to errors and warnings.

Additional instructions...

>Start the app.
>
>Visit http://localhost:4000/
>
>yarn start

... mislead the user to first open the server address in a browser and then start the server.

## Importance

Examples should provide users with working solution without any problems. Especially examples as basic as this one.

## Related

graphql-server-typeorm-postgres does not work either. I did not try other examples but I doubt they work.

## git clone result

```bash
prishub@ubuntu-one:~/dev/xxx/web/accounts-js$ git clone https://github.com/accounts-js/accounts.git
Cloning into 'accounts'...
remote: Enumerating objects: 32, done.
remote: Counting objects: 100% (32/32), done.
remote: Compressing objects: 100% (20/20), done.
remote: Total 15721 (delta 14), reused 20 (delta 7), pack-reused 15689
Receiving objects: 100% (15721/15721), 7.63 MiB | 2.10 MiB/s, done.
Resolving deltas: 100% (11142/11142), done.
prishub@ubuntu-one:~/dev/xxx/web/accounts-js$ cd accounts/examples/graphql-server-typescript/
```

## yarn install result

```bash
prishub@ubuntu-one:~/dev/xxx/web/accounts-js/accounts/examples/graphql-server-typescript$ yarn install
yarn install v1.17.3
[1/4] Resolving packages...
warning Resolution field "graphql-tools@4.0.5" is incompatible with requested version "graphql-tools@4.0.4"
warning Resolution field "graphql-tools@4.0.5" is incompatible with requested version "graphql-tools@4.0.4"
warning Resolution field "@types/node@12.7.4" is incompatible with requested version "@types/node@^10.1.0"
[2/4] Fetching packages...
info fsevents@2.0.7: The platform "linux" is incompatible with this module.
info "fsevents@2.0.7" is an optional dependency and failed compatibility check. Excluding it from installation.
info fsevents@1.2.9: The platform "linux" is incompatible with this module.
info "fsevents@1.2.9" is an optional dependency and failed compatibility check. Excluding it from installation.
[3/4] Linking dependencies...
warning " > graphql-tools@4.0.5" has unmet peer dependency "graphql@^0.13.0 || ^14.0.0".
warning "graphql-tools > apollo-link@1.2.12" has unmet peer dependency "graphql@^0.11.3 || ^0.12.3 || ^0.13.0 || ^14.0.0".
warning "graphql-tools > apollo-utilities@1.3.2" has unmet peer dependency "graphql@^0.11.0 || ^0.12.0 || ^0.13.0 || ^14.0.0".
warning " > ts-jest@24.0.2" has unmet peer dependency "jest@>=24 <25".
warning " > @accounts/express-session@0.19.0" has unmet peer dependency "express@^4.16.3".
warning " > @accounts/express-session@0.19.0" has unmet peer dependency "express-session@^1.15.6".
warning " > @accounts/graphql-api@0.19.0" has unmet peer dependency "@graphql-modules/core@0.7.9".
warning " > @accounts/graphql-api@0.19.0" has unmet peer dependency "graphql-tag@^2.10.0".
warning "workspace-aggregator-58e386bf-df20-49ab-83f3-a64598680545 > @accounts/graphql-api > graphql-toolkit@0.5.12" has unmet peer dependency "graphql-tag-pluck@^0.8.3".
warning "workspace-aggregator-58e386bf-df20-49ab-83f3-a64598680545 > @accounts/graphql-client > graphql-tag@2.10.1" has unmet peer dependency "graphql@^0.9.0 || ^0.10.0 || ^0.11.0 || ^0.12.0 || ^0.13.0 || ^14.0.0".
warning "workspace-aggregator-58e386bf-df20-49ab-83f3-a64598680545 > @examples/react-graphql-typescript > react-apollo@2.5.8" has unmet peer dependency "apollo-client@^2.6.3".
warning "workspace-aggregator-58e386bf-df20-49ab-83f3-a64598680545 > @accounts/boost > @graphql-modules/core > @graphql-modules/di@0.7.11" has unmet peer dependency "reflect-metadata@^0.1.12".
warning "workspace-aggregator-58e386bf-df20-49ab-83f3-a64598680545 > @examples/react-graphql-typescript > react-scripts > @typescript-eslint/eslint-plugin@1.13.0" has incorrect peer dependency "eslint@^5.0.0".
warning "workspace-aggregator-58e386bf-df20-49ab-83f3-a64598680545 > @examples/react-graphql-typescript > react-scripts > @typescript-eslint/parser@1.13.0" has incorrect peer dependency "eslint@^5.0.0".
[4/4] Building fresh packages...
Done in 253.00s.
```

## start attempt result

```bash
prishub@ubuntu-one:~/dev/xxx/web/accounts-js/accounts/examples/graphql-server-typescript$ yarn start
yarn run v1.17.3
$ NODE_ENV=development nodemon -w src -x ts-node src/index.ts
[nodemon] 1.19.2
[nodemon] to restart at any time, enter `rs`
[nodemon] watching dir(s): src/**/*
[nodemon] starting `ts-node src/index.ts`

/home/prishub/dev/xxx/web/accounts-js/accounts/node_modules/ts-node/src/index.ts:245
    return new TSError(diagnosticText, diagnosticCodes)
           ^
TSError: ⨯ Unable to compile TypeScript:
src/index.ts:1:33 - error TS2307: Cannot find module '@accounts/database-manager'.

1 import { DatabaseManager } from '@accounts/database-manager';
                                  ~~~~~~~~~~~~~~~~~~~~~~~~~~~~
src/index.ts:2:32 - error TS2307: Cannot find module '@accounts/graphql-api'.

2 import { AccountsModule } from '@accounts/graphql-api';
                                 ~~~~~~~~~~~~~~~~~~~~~~~
src/index.ts:3:30 - error TS2307: Cannot find module '@accounts/mongo'.

3 import MongoDBInterface from '@accounts/mongo';
                               ~~~~~~~~~~~~~~~~~
src/index.ts:4:34 - error TS2307: Cannot find module '@accounts/password'.

4 import { AccountsPassword } from '@accounts/password';
                                   ~~~~~~~~~~~~~~~~~~~~
src/index.ts:5:32 - error TS2307: Cannot find module '@accounts/server'.

5 import { AccountsServer } from '@accounts/server';
                                 ~~~~~~~~~~~~~~~~~~

    at createTSError (/home/prishub/dev/xxx/web/accounts-js/accounts/node_modules/ts-node/src/index.ts:245:12)
    at reportTSError (/home/prishub/dev/xxx/web/accounts-js/accounts/node_modules/ts-node/src/index.ts:249:19)
    at getOutput (/home/prishub/dev/xxx/web/accounts-js/accounts/node_modules/ts-node/src/index.ts:362:34)
    at Object.compile (/home/prishub/dev/xxx/web/accounts-js/accounts/node_modules/ts-node/src/index.ts:395:32)
    at Module.m._compile (/home/prishub/dev/xxx/web/accounts-js/accounts/node_modules/ts-node/src/index.ts:473:43)
    at Module._extensions..js (internal/modules/cjs/loader.js:947:10)
    at Object.require.extensions.<computed> [as .ts] (/home/prishub/dev/xxx/web/accounts-js/accounts/node_modules/ts-node/src/index.ts:476:12)
    at Module.load (internal/modules/cjs/loader.js:790:32)
    at Function.Module._load (internal/modules/cjs/loader.js:703:12)
    at Function.Module.runMain (internal/modules/cjs/loader.js:999:10)
```

**Comment by bbenoist** [comment]
It works on my machine ™️ (macOS) when using `npm install && npm run start`.
However, I have the same problem with `yarn && yarn start`.

**Comment by pradel** [comment]
It's because you need to run the lerna bootsrap script first with yarn. The documentation need to be updated to reflect that :)

**Comment by prishub** [comment]
> It's because you need to run the lerna bootsrap script first with yarn. The documentation need to be updated to reflect that :)

I am not quite sure I understand what you want to say. Let’s assume I am a beginner.

Typically, I am a developer and I want to use some library. I start a new project and I install the dependencies of the library I want to use. These were published by its authors and I use a package manager to get them.  Then I fill in some code and I should be able to build and run what I wrote.

To learn how to use new stuff and to get inspired, I want to download some pre-made solution, like an example. Such an example should show me how my application would look like should I use the library. This is what examples are for, right? 

Example usually has its dependencies already listed in package.json. I assume I just install these dependencies using package.json and all the necessary dependencies should get installed as if they have been added by me during my application development process and in the end the example should run.

Sometimes such an example is published in its own repository, sometimes it is provided as a part of a library repository. Like it is here. However, such a library repository is setup for development of a library itself not a development of applications that are supposed to only use the deliverables of a library.

As a developer who wants to learn about a new library and use it in his applications I am not interested in development of the library, its monorepo setup, workspaces, packages inside it, bootstrapping, linking, whatever, I just want to use its deliverables as if I did not have the library repository cloned on my local drive. I only cloned it to get the example folder, I don’t care about the rest.

Are you actually telling me that I cannot use that example as a user who just copied files and folders, installed modules based on package.json, and that I can only use it if I act as a developer of the library itself, i.e. I clone its repository, do some library development related steps and then it eventually runs? But then developing of my own application I will most likely have to do the same, literally give up my own configuration and actually integrate it with your library. Or am I completely lost?
